src/main/java/frc/robot/commands/Turret/VisionAimTurret.java

```diff
@@ -38,25 +52,47 @@ public void initialize() {
   @Override
   public void execute() {
     target = -vision.limelight.getOffsetX() + turret.getTurretAngle();
+    SmartDashboard.putNumber("oldTargetPosition", oldTargetPosition);
 
-    if (vision.limelight.hasTarget()) {
-      turret.setTurretAngle(target);
-    }
+    boolean isPressed = RobotContainer.coDriverStick.btn_A.get();
 
-    // if (shooter.isGoalHighHub()) {
-    // shooter.setGoalRPM(vision.getIdealMediumHoodRPM());
-    // } else {
-    // shooter.setGoalRPM(vision.getIdealLowerHubRPM());
-    // }
+    if (isPressed) {
+      if (vision.limelight.hasTarget()) {
+        if (!is360Spin) {
+          turret.setTurretAngle(target);
+          oldNavXPosition = navX.navx.getYaw();
+        }
+        oldTargetPosition = target;
+      } else {
+        changeInNavx = navX.navx.getYaw() - oldNavXPosition;
+
+        newTargetPosition = oldTargetPosition + changeInNavx;
+        SmartDashboard.putNumber("newTargetPosition", newTargetPosition);
 
-    shooter.setGoalRPM(vision.getIdealMediumHoodRPM());
+        if (newTargetPosition < RobotPreferences.TurretPrefs.turretMinAngleDegrees.getValue()) {
+          is360Spin = true;
+          oppositePosition = RobotPreferences.TurretPrefs.turretMinAngleDegrees.getValue() - newTargetPosition;
+          turret.setTurretAngle(RobotPreferences.TurretPrefs.turretMaxAngleDegrees.getValue() - oppositePosition);
+          is360Spin = false;
```

Comment:
`is360spin` is set to true and immediately set to false within the same block (curly brackets). This makes it seem as if nothing ever changed to the code. Would this still be useful given that fact?